internal/provider/data_source_secure_connect_bundle_url.go

```diff
@@ -40,18 +42,36 @@ func dataSourceSecureConnectBundleURLRead(ctx context.Context, d *schema.Resourc
 
 	databaseID := d.Get("database_id").(string)
 
-	resp, err := client.GenerateSecureBundleURLWithResponse(ctx, astra.DatabaseIdParam(databaseID))
-	if err != nil {
-		return diag.FromErr(err)
-	}
+	var credsURL *astra.CredsURL
+	if err := resource.RetryContext(ctx, time.Minute, func() *resource.RetryError {
+		resp, err := client.GenerateSecureBundleURLWithResponse(ctx, astra.DatabaseIdParam(databaseID))
+		if err != nil || resp.StatusCode() >= 500 {
+			return resource.RetryableError(err)
+		}
+
+		// 409 Conflict can be returned if the database is not yet ready
+		if resp.JSON409 != nil {
+			return resource.RetryableError(fmt.Errorf("cannot create secure bundle url: %s", string(resp.Body)))
+		}
 
-	bundleURL := resp.JSON200
-	if bundleURL == nil {
-		return diag.Errorf("failed to get secure connect bundle: %s", string(resp.Body))
+		// Any other 400 status code is not retried
+		if resp.StatusCode() >= 400 {
+			return resource.NonRetryableError(fmt.Errorf("error trying to create secure bundle url: %s", string(resp.Body)))
+		}
+
+		// Any response other than 200 is unexpected
+		credsURL := resp.JSON200
```

Comment:
Shouldn't this be `=` instead of `:=` since we want to use the var up on line 45?

Reply:
Hah you found the error I ran into during the demo!